File: Go/EX10.go
package main

import "fmt"

// Definindo a estrutura Calculadora
type Calculadora struct{}

// Função somar para dois números
func (c Calculadora) SomarDois(num1, num2 int) int {
    return num1 + num2
}

// Função somar para três números
func (c Calculadora) SomarTres(num1, num2, num3 int) int {
    return num1 + num2 + num3
}

func main() {
    // Criando um objeto da Calculadora
    calc := Calculadora{}

    // Soma de dois números
    resultado1 := calc.SomarDois(10, 20)
    fmt.Printf("Soma de 10 + 20: %d\n", resultado1)

    // Soma de três números
    resultado2 := calc.SomarTres(10, 20, 30)
    fmt.Printf("Soma de 10 + 20 + 30: %d\n", resultado2)
}

File: Java/EX10.java
class Calculadora {

    // Método somar para dois números
    public int somar(int num1, int num2) {
        return num1 + num2;
    }

    // Método somar para três números
    public int somar(int num1, int num2, int num3) {
        return num1 + num2 + num3;
    }

    public static void main(String[] args) {
        // Criando um objeto da Calculadora
        Calculadora calc = new Calculadora();

        // Soma de dois números
        int resultado1 = calc.somar(10, 20);
        System.out.println("Soma de 10 + 20: " + resultado1);

        // Soma de três números
        int resultado2 = calc.somar(10, 20, 30);
        System.out.println("Soma de 10 + 20 + 30: " + resultado2);
    }
}


